app/scripts/views/mixins/modal-settings-panel-mixin.js

```diff
@@ -57,6 +57,10 @@ define(function (require, exports, module) {
         this._returnToClients();
         break;
       case 'settings/two_step_authentication/recovery_codes':
+        if (this.model.get('previousViewName') === 'sign_in_recovery_code') {
```

Comment:
Updating the modal-settings-panel-mixin for logic that should be in the view itself is killing me. I tried to do a quick PR to push all the logic in this method back to the views today, and failed.